### PR TITLE
Fixes get_signal_engagement_by_name call in signals create module and several type annotations

### PR DESCRIPTION
**Pull request by mvilanova**
This pull request includes several changes across multiple files to improve type annotations and import organization. The most important changes involve updating function signatures to include type annotations and reorganizing imports for better readability.

### Type Annotation Improvements:

* [`src/dispatch/data/alert/service.py`](diffhunk://#diff-65fc2ede653b957640f11342b83c3531e379bba33cb52e3d58e03d6c20c5276aL19-R20): Updated the `get_by_name_or_raise` function to include type annotations for the `alert_in` parameter.
* [`src/dispatch/organization/service.py`](diffhunk://#diff-8f2e2a892226548555aea19e5aef00ad1a93cc06ff6810ddf219587113851a99L47-R47): Updated multiple functions (`get_by_name_or_raise`, `get_by_slug_or_raise`, `get_by_name_or_default`) to include type annotations for the `organization_in` parameter. [[1]](diffhunk://#diff-8f2e2a892226548555aea19e5aef00ad1a93cc06ff6810ddf219587113851a99L47-R47) [[2]](diffhunk://#diff-8f2e2a892226548555aea19e5aef00ad1a93cc06ff6810ddf219587113851a99L70-R70) [[3]](diffhunk://#diff-8f2e2a892226548555aea19e5aef00ad1a93cc06ff6810ddf219587113851a99L88-R88)
* [`src/dispatch/project/service.py`](diffhunk://#diff-d3b50429486d5381e604b731c1c849675cc68a85a9b97326897e6b63af26d20fL45-R45): Updated the `get_by_name_or_raise` and `get_by_name_or_default` functions to include type annotations for the `project_in` parameter. [[1]](diffhunk://#diff-d3b50429486d5381e604b731c1c849675cc68a85a9b97326897e6b63af26d20fL45-R45) [[2]](diffhunk://#diff-d3b50429486d5381e604b731c1c849675cc68a85a9b97326897e6b63af26d20fL63-R63)
* [`src/dispatch/service/service.py`](diffhunk://#diff-cd8092346b3d8faf5cd2a63ba3d756a875652669bc8b55ece2797002171e352aL34-R34): Updated the `get_by_name_or_raise` function to include type annotations for the `service_in` parameter.
* [`src/dispatch/signal/service.py`](diffhunk://#diff-c88619ded82e7ac0ad4a369ff984ebc366b0c766cd88e7313bb54c49fdaff8f1L116-R118): Updated multiple functions (`get_signal_engagement_by_name_or_raise`, `get_signal_filter_by_name_or_raise`) to include type annotations for the `signal_engagement_in` and `signal_filter_in` parameters respectively. [[1]](diffhunk://#diff-c88619ded82e7ac0ad4a369ff984ebc366b0c766cd88e7313bb54c49fdaff8f1L116-R118) [[2]](diffhunk://#diff-c88619ded82e7ac0ad4a369ff984ebc366b0c766cd88e7313bb54c49fdaff8f1L229-R230)

### Import Organization:

* [`src/dispatch/data/alert/service.py`](diffhunk://#diff-65fc2ede653b957640f11342b83c3531e379bba33cb52e3d58e03d6c20c5276aR2-R7): Reorganized imports to group similar imports together and removed redundant imports.
* [`src/dispatch/project/service.py`](diffhunk://#diff-d3b50429486d5381e604b731c1c849675cc68a85a9b97326897e6b63af26d20fL5-R10): Reorganized imports to group similar imports together and removed redundant imports.
* [`src/dispatch/service/service.py`](diffhunk://#diff-cd8092346b3d8faf5cd2a63ba3d756a875652669bc8b55ece2797002171e352aL11-R11): Reorganized imports to group similar imports together and removed redundant imports.
* [`src/dispatch/tag/service.py`](diffhunk://#diff-4fa60ad7453a9f9e85b227037e328e06c5d25514c7be9ac3c52234a770470d2dR2-R9): Reorganized imports to group similar imports together and removed redundant imports.
* [`src/dispatch/tag_type/service.py`](diffhunk://#diff-3b2dd5a23db9667d0fc3f57c5c62b2d2d4e39d8354ac566b760dc1da8dd86224L4-R5): Reorganized imports to group similar imports together and removed redundant imports.
* [`src/dispatch/workflow/service.py`](diffhunk://#diff-83474aa583cabdcdcb3a7bba06885bf62e93654357a1ea47e40024ccc06ad7acR3-L7): Reorganized imports to group similar imports together and removed redundant imports. [[1]](diffhunk://#diff-83474aa583cabdcdcb3a7bba06885bf62e93654357a1ea47e40024ccc06ad7acR3-L7) [[2]](diffhunk://#diff-83474aa583cabdcdcb3a7bba06885bf62e93654357a1ea47e40024ccc06ad7acL21-R25)

### CODE REVIEW

File: src/dispatch/signal/service.py [hunk 1]
```diff
@@ -371,9 +372,9 @@ def create(*, db_session: Session, signal_in: SignalCreate) -> Signal:
     signal.entity_types = entity_types
 
     engagements = []
-    for eng in signal_in.engagements:
+    for signal_engagement_in in signal_in.engagements:
         signal_engagement = get_signal_engagement_by_name(
```
Comment (mvilanova) [inline src/dispatch/signal/service.py:376]: We were not calling the function correctly.

File: src/dispatch/signal/service.py [hunk 2]
```diff
@@ -371,9 +372,9 @@ def create(*, db_session: Session, signal_in: SignalCreate) -> Signal:
...
-            db_session=db_session, project_id=project.id, signal_engagement_in=eng
+            db_session=db_session, project_id=project.id, name=signal_engagement_in.name
```
Comment (wssheldon) [inline src/dispatch/signal/service.py:377]: Is this the bug?

**Review by wssheldon — APPROVED** [review]

File: src/dispatch/signal/service.py (hunk 2 above)
Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: I'm not 100% sure as this is inside the create function, but I didn't find any issues with the update one.

Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: I think I found the issue in update. We're actually not updating the engagement if we find it: https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/service.py#L456

Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: We should update the engagement here if we found it https://github.com/Netflix/dispatch/blob/master/src/dispatch/signal/service.py#L461

Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: I think we have the same issue with entity types, filters, and workflows...

Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: We don't update them before appending...

Comment (mvilanova) [reply src/dispatch/signal/service.py:377]: I'll merge this PR and create another one on Monday.

Comment (wssheldon) [reply src/dispatch/signal/service.py:377]: Sounds good, thanks for tracking it down

**Pull request merged**